src/wwwroot/app/services/thesaurusService.js

```diff
@@ -0,0 +1,124 @@
+import { has, find, concat, remove, filter, map, forEach } from 'lodash';
+
+import { THESAURUS_STRUCTURES } from './ThesaurusStructuresStore.js';
+
+let _HttpService, _$q;
+
+const cache = { };
+
+export default class ThesaurusService {
+   constructor(HttpService, $q) {
+      'ngInject';
+      _HttpService = HttpService;
+      _$q = $q;
+   }
+
+   getThesaurusTopics(thesaurusName) {
+      if (!has(cache, thesaurusName)) {
```

Comment:
I really dislike `not` in condition.

please, replace it onto

```
if (has(cache, thesaurusName)) {
   return _$q.when(cache[thesaurusName]);
}
....
```

this will be much better